include/constants/item_config.h

```diff
@@ -0,0 +1,33 @@
+#ifndef GUARD_CONSTANTS_ITEM_CONFIG_H
+#define GUARD_CONSTANTS_ITEM_CONFIG_H
+
+// Used by other branches to communicate with each other.
+#define ITEM_EXPANSION
+
+#ifndef GEN_3
+#define GEN_3 0
+#define GEN_4 1
+#define GEN_5 2
+#define GEN_6 3
+#define GEN_7 4
+#define GEN_8 5
+#endif
+
+// Item config
+#define P_SHINY_CHARM_REROLLS   3       // Amount of re-rolls if the player has the Shiny Charm. Set to 0 to disable Shiny Charm's effects.
```

Comment:
why the `P_` prefix? if battle engine config prefixes are `B_` shouldn't item prefixes by `I_`?

Reply:
You're right, I feel the prefix for all of these should be `I_`, as they're options on Item Expansion. (including the ball ones)